core/server/master/src/main/java/alluxio/master/file/loadmanager/LoadManager.java

```diff
@@ -0,0 +1,310 @@
+        LoadInfo info = mLoadPathToInfo.get(path);
+        /*Only update bandwidth here.*/
+        long newBandWidth = options.getBandwidth();
+        info.getLoadOptions().setBandwidth(newBandWidth);
+        return ValidationStatus.Update_Options;
+      }
+      return ValidationStatus.Ignored;
+    }
+    return ValidationStatus.New;
+  }
+
+  static class Scheduler {
+    private static final int CAPACITY = 100;
+    private static final long TIMEOUT = 100;
+    private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
+    private final BlockingQueue<Load> mLoadQueue = new LinkedBlockingQueue<>(CAPACITY);
+    private final AtomicInteger mCurrentSize = new AtomicInteger();
+    private final AtomicLong mIdGenerator = new AtomicLong();
+
+    void schedule(Load load)
+            throws ResourceExhaustedException, InterruptedException {
+      if (mCurrentSize.get() == CAPACITY) {
+        throw new ResourceExhaustedException(
+                "Insufficient capacity to enqueue load tasks!");
+      }
+
+      boolean offered = mLoadQueue.offer(load, TIMEOUT, TimeUnit.MILLISECONDS);
+      if (offered) {
+        mCurrentSize.incrementAndGet();
+      } else {
+        LOG.warn("Cannot enqueue load to the queue, may lose track on this load!"
+                + load.getDetailedInfo());
+      }
+
+      mExecutorService.submit(() -> {
+        try {
+          runLoad(load);
+        } catch (AlluxioRuntimeException e) {
+          handleErrorOnStatuses(); // handle based on status
+        } catch (TimeoutException e) {
+          // add retry and handle timeout caused by checking available workers
+        }
+      });
+    }
+
+    void runLoad(Load load) throws AlluxioRuntimeException, TimeoutException {
+      BlockIterator<Long> blockIterator = new BlockIterator<>(load.getPath());
+
+      while (blockIterator.hasNextBatch()) {
+        // Get a worker to handle the load task.
+        ExecutionWorkerInfo worker = getNextAvailableWorker();
+        if (worker == null) { // if no workers available, continue
+          continue;
+        }
+
+        List<Long> blockIds = blockIterator.getNextBatchBlocks();
+        BlockBatch blockBatch = new BlockBatch(blockIds, getNextBatchId());
+        load.addBlockBatch(blockBatch);
+        LoadResponse status = worker.execute(blockBatch);
+        handleCompletion(status);
+      }
+    }
+
+    void handleCompletion(LoadResponse loadResponse) {
+    }
+
+    void handleErrorOnStatuses() {
+    }
+
+    private long getNextBatchId() {
+      return mIdGenerator.incrementAndGet();
+    }
+
+    private static ExecutionWorkerInfo getNextAvailableWorker() throws TimeoutException {
+      // update currently available workers and get a next available worker.
+      return null;
+    }
+  }
+
+  static class Load {
+    private final long mLoadId;
+    private final String mPath;
+    private final List<BlockBatch> mBlockBatches = Lists.newArrayList();
```

Comment:
Create an immutable list outside of this class and initialize in constructor, rather than use `addBlockBatch`

Reply:
This is for adding new block batch while iterating/listing that list of blocks and call worker to run, so we should have this `addBlockBatch` to add that list of blocks, right?

Reply:
You can add to a list (ImmutableList.builder) before constructing `Load` rather than add each element with an `addBlockBatch` API.

Reply:
Discussed offline, will submit this part in this PR as is for now